### PR TITLE
Update project structure

### PR DESCRIPTION
**Pull request by demurgos**
This PR updates the directory structure with the latest [generator](https://github.com/typings/generator-typings) structure. It does not change the definitions.

**Note**: The generator-typings structure does not support `mocha` unit-tests so they were removed.

Reference:
https://github.com/types/npm-vinyl/pull/2#issuecomment-254422414
https://github.com/types/npm-vinyl/pull/3

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,15 +4,32 @@
...
+    "build": "echo building... && typings bundle -o out/index.d.ts",
+    "lint": "echo linting... && tslint \"**/*.ts\" -e \"source/**\" -e \"source-test/**\" -e \"out/**\" -e \"node_modules/**\" -e \"typings/**\"",
+    "browser-test": "echo running browser tests... && echo no browser test",
+    "source-test": "echo running source tests... && echo no source test",
+    "test": "echo running server tests... && cd test && ts-node ../node_modules/blue-tape/bin/blue-tape \"**/*.ts\" | tap-spec",
+    "watch": "onchange -w \"**/*.ts\" -i -e \"out/**\" -- npm -s run build+test",
+    "publish": "npm -s run lint+build+test && echo please publish to typings/registry",
+    "all-tests": "npm test && npm run source-test",
+    "build+test": "npm run build && npm run all-tests",
+    "lint+build+test": "npm run lint && npm run build+test",
```
Comment (felixfbecker) [inline package.json]: Just keep the old test scripts and keep the tests. Removing the tests is a no-go

**Comment by demurgos** [comment]
I restored the original tests. But just to be sure, you never run the tests: you just compile them ?

(It also seems that I don't have the same `tsc` version as Travis, I'll check why it failed)

**Comment by demurgos** [comment]
Ok, the latest typings for `Node` require `typescript` 2. I'll revert the Node typings to an older version for the moment.

**Comment by demurgos** [comment]
It should be OK now, could you see if this is fine ?

**Comment by felixfbecker** [comment]
@demurgos Yes, the file is just compiled. Please just leave that as-is right now, converting them all to runtime tests is out of scope for this PR. That means please also leave out the dependencies on blue-tape. Upgrading to TS2 is not a problem.

**Comment by demurgos** [comment]
I took me some time, but I think that it should be good: the dependencies are updated (use TS2, no more blue-tape) and there was no breaking-change with the previous typings.

**Pull request merged**